src/hooks/useTrackPageState.ts

```diff
@@ -1,126 +1,104 @@
-import { useState } from 'react';
-import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
-import {
-  fetchTracks,
-  fetchGenres,
-  deleteTrack,
-  updateTrack,
-  createTrack,
-} from '../services/api/tracks';
-import useDebounce from './useDebounce.ts';
+import { useEffect, useState } from 'react';
 import { SelectChangeEvent } from '@mui/material';
+import { setParam, getParam } from '../services/api/urlParams.ts';
+import { O, pipe } from '@mobily/ts-belt';
+import { useDebounce } from '../hooks';
+
+import { useTracksQuery, useGenresQuery } from '../hooks';
+
+import {
+  useDeleteTrackMutation,
+  useDeleteMultipleTracksMutation,
+  useCreateTrackMutation,
+  useUpdateTrackMutation,
+} from '../hooks';
 
 const useTrackPageState = () => {
-  const queryClient = useQueryClient();
-  const [page, setPage] = useState(1);
+  const rawPageParam = getParam('page');
+  const rawSortParam = getParam('sort');
+  const rawGenreParam = getParam('genre');
+  const rawArtistParam = getParam('artist');
+  const rawSearchParam = getParam('search');
+
+  const [page, setPage] = useState<number>(() =>
```

Comment:
Do we still need useState in `useTrackPageState`? Since every parameter is saved in query parameters in url it seems like state isn't needed. Let me know what you think about this